utils/src/main/java/software/amazon/awssdk/utils/cache/NonBlocking.java

```diff
@@ -37,80 +46,204 @@
+     */
+    private static final ThreadPoolExecutor EXECUTOR =
+        new ThreadPoolExecutor(0, Integer.MAX_VALUE,
+                               60L, TimeUnit.SECONDS,
+                               new SynchronousQueue<>(),
+                               new ThreadFactoryBuilder().daemonThreads(true).build());
+
+    /**
+     * An incrementing number, used to uniquely identify an instance of NonBlocking in the {@link #asyncThreadName}.
+     */
+    private static final AtomicLong THREAD_NUMBER = new AtomicLong(0);
+
     /**
      * Whether we are currently refreshing the supplier. This is used to make sure only one caller is blocking at a time.
      */
-    private final AtomicBoolean currentlyRefreshing = new AtomicBoolean(false);
+    private final AtomicBoolean currentlyPrefetching = new AtomicBoolean(false);
 
     /**
-     * How frequently to automatically refresh the supplier in the background.
+     * Name of the thread refreshing the cache for this strategy.
      */
-    private final Duration asyncRefreshFrequency;
+    private final String asyncThreadName;
 
     /**
-     * Single threaded executor to asynchronous refresh the value.
+     * The refresh task currently scheduled for this non-blocking instance. We ensure that no more than one task is scheduled
+     * per instance.
      */
-    private final ScheduledExecutorService executor;
+    private final AtomicReference<ScheduledFuture<?>> refreshTask = new AtomicReference<>();
+
+    /**
+     * The minimum amount of time allowed between async refreshes, primarily adjustable for testing purposes.
+     */
+    private final Duration minimumRefreshFrequency;
+
+    /**
+     * Whether this strategy has been shutdown (and should stop doing background refreshes)
+     */
+    private volatile boolean shutdown = false;
+
+    /**
+     * The cached supplier using this non-blocking instance.
+     */
+    private volatile CachedSupplier<?> cachedSupplier;
+
+    static {
+        // Ensure that cancelling a task actually removes it from the queue.
+        SCHEDULER.setRemoveOnCancelPolicy(true);
+    }
 
     /**
      * Create a non-blocking prefetch strategy that uses the provided value for the name of the background thread that will be
      * performing the update.
      */
     public NonBlocking(String asyncThreadName) {
-        this(asyncThreadName, Duration.ofMinutes(1));
+        this(asyncThreadName, Duration.ofSeconds(60));
     }
 
     @SdkTestInternalApi
-    NonBlocking(String asyncThreadName, Duration asyncRefreshFrequency) {
-        this.executor = newExecutor(asyncThreadName);
-        this.asyncRefreshFrequency = asyncRefreshFrequency;
+    NonBlocking(String asyncThreadName, Duration minimumRefreshFrequency) {
+        this.asyncThreadName = asyncThreadName + "-" + THREAD_NUMBER.getAndIncrement();
+        this.minimumRefreshFrequency = minimumRefreshFrequency;
```

Comment:
Nice, like that this previous 'polling' duration is now the minimum refresh frequency.